internal/btf/core.go

```diff
@@ -15,10 +16,18 @@ import (
 // Code in this file is derived from libbpf, which is available under a BSD
 // 2-Clause license.
 
+type OptionalLocalValue int64
```

Comment:
Yeah I wasn't very happy with it. I thought about adding a `COREKind`, but since these are constants coming from BTF I felt icky adding a "magic" non-overlapping kind there, e.g. to me it felt worse than doing it this way.

The validation does make sense as often we can figure out what we're expecting to see, so it's good to keep it.

Reply:
We can introduce a separate type like`FixupKind` if we want and unexport current `COREKind`. If I remember correctly Fixup.Kind is only used in `String()`.

Reply:
Hm, not quite following what you're proposing. What would this `FixupKind` look like and why would we want to unexport `COREKind`?

Reply:
Think of `COREKind` as defined by libbpf, more specifically by the C CORE headers. `FixupKind` would be a superset of COREKind, with additional types that are useful to us. I'd unexport COREKind since it's kind of an implementation detail of libbpf, and to avoid confusion between COREKind and FixupKind.

Reply:
Yeah, you did a good job with a bad idea of mine :D At that point I hadn't realised that it's necessary to mix bitfields and non-bitfields in the same fixup. We should probably chuck the validate bool in the Fixup, but that is something for another day. Unexporting coreKind is definitely the way to go though, we shouldn't expose libbpf ABI if we can avoid it.